ksqldb-engine/src/test/java/io/confluent/ksql/planner/plan/PullFilterNodeTest.java

```diff
@@ -896,15 +933,30 @@ public void shouldThrowMultiKeyExpressionsThatDontCoverAllKeys() {
             expression,
             metaStore,
             ksqlConfig,
-            false
-        ));
+            false,
+            plannerOptions));
 
     // Then:
     assertThat(e.getMessage(), containsString(
         "Multi-column sources must specify every key in the WHERE clause. "
             + "Specified: [`K1`] Expected: [`K1` INTEGER KEY, `K2` INTEGER KEY]"));
   }
 
+  @Test
+  public void shouldExtractConstraintForMultiKeyExpressionsThatDontCoverAllKeys_tableScan() {
+    // Given:
+    when(plannerOptions.getTableScansEnabled()).thenReturn(true);
+    when(source.getSchema()).thenReturn(MULTI_KEY_SCHEMA);
+    final Expression expression = new ComparisonExpression(
+        Type.EQUAL,
+        new UnqualifiedColumnReferenceExp(ColumnName.of("K1")),
+        new IntegerLiteral(1)
+    );
+
+    // Then:
+    expectTableScan(expression, false);
```

Comment:
I'm a bit confused, shouldn't table scan be expected in this case?

Reply:
This reads a little funny, the false parameter is `isWindowed`.